dronegen/buildbox.go

```diff
@@ -35,46 +35,66 @@ func buildboxPipelineSteps() []step {
-			"PROD_AWS_SECRET_ACCESS_KEY":    {fromSecret: "PRODUCTION_BUILDBOX_DRONE_USER_ECR_SECRET"},
+	assumeStagingRoleStep := kubernetesAssumeAwsRoleStep(kubernetesRoleSettings{
+		awsRoleSettings: awsRoleSettings{
+			awsAccessKeyID:     value{fromSecret: "STAGING_BUILDBOX_DRONE_USER_ECR_KEY"},
+			awsSecretAccessKey: value{fromSecret: "STAGING_BUILDBOX_DRONE_USER_ECR_SECRET"},
+			role:               value{fromSecret: "STAGING_BUILDBOX_DRONE_ECR_AWS_ROLE"},
 		},
-		Volumes: []volumeRef{volumeRefDocker},
-		Commands: []string{
-			`apk add --no-cache make aws-cli`,
-			`chown -R $UID:$GID /go`,
-			// Authenticate to staging registry
-			`export AWS_ACCESS_KEY_ID="$STAGING_AWS_ACCESS_KEY_ID"`,
-			`export AWS_SECRET_ACCESS_KEY="$STAGING_AWS_SECRET_ACCESS_KEY"`,
-			`aws ecr get-login-password --region=us-west-2 | docker login -u="AWS" --password-stdin ` + StagingRegistry,
-			// Build buildbox image
-			fmt.Sprintf(`make -C build.assets %s`, buildboxName),
-			// Retag for staging registry
-			fmt.Sprintf(`docker tag %s/gravitational/teleport-%s:$BUILDBOX_VERSION %s/gravitational/teleport-%s:$BUILDBOX_VERSION-$DRONE_COMMIT_SHA`, ProductionRegistry, buildboxName, StagingRegistry, buildboxName),
-			// Push to staging registry
-			fmt.Sprintf(`docker push %s/gravitational/teleport-%s:$BUILDBOX_VERSION-$DRONE_COMMIT_SHA`, StagingRegistry, buildboxName),
-			// Authenticate to production registry
-			`docker logout ` + StagingRegistry,
-			`export AWS_ACCESS_KEY_ID="$PROD_AWS_ACCESS_KEY_ID"`,
-			`export AWS_SECRET_ACCESS_KEY="$PROD_AWS_SECRET_ACCESS_KEY"`,
-			`aws ecr-public get-login-password --region=us-east-1 | docker login -u="AWS" --password-stdin ` + ProductionRegistry,
-			// Push to production registry
-			fmt.Sprintf(`docker push %s/gravitational/teleport-%s:$BUILDBOX_VERSION`, ProductionRegistry, buildboxName),
+		configVolume: volumeRefAwsConfig,
+	})
+	assumeStagingRoleStep.Name = fmt.Sprintf("Assume Staging %s AWS Role", buildboxName)
+	assumeProductionRoleStep := kubernetesAssumeAwsRoleStep(kubernetesRoleSettings{
+		awsRoleSettings: awsRoleSettings{
+			awsAccessKeyID:     value{fromSecret: "PRODUCTION_BUILDBOX_DRONE_USER_ECR_KEY"},
+			awsSecretAccessKey: value{fromSecret: "PRODUCTION_BUILDBOX_DRONE_USER_ECR_SECRET"},
+			role:               value{fromSecret: "PRODUCTION_BUILDBOX_DRONE_ECR_AWS_ROLE"},
+		},
+		configVolume: volumeRefAwsConfig,
+	})
+	assumeProductionRoleStep.Name = fmt.Sprintf("Assume Production %s AWS Role", buildboxName)
+	return []step{
+		assumeStagingRoleStep,
+		step{
+			Name:    fmt.Sprintf("Build %s and push to Staging", buildboxName),
+			Image:   "docker",
+			Volumes: []volumeRef{volumeRefDocker, volumeRefAwsConfig},
+			Commands: []string{
+				`apk add --no-cache make aws-cli`,
+				`chown -R $UID:$GID /go`,
+				// Authenticate to staging registry
+				`aws ecr get-login-password --region=us-west-2 | docker login -u="AWS" --password-stdin ` + StagingRegistry,
```

Comment:
I understand this is existing stuff, but this caught my eye. Is it expected that staging is `us-west-2` but production below is `us-east-1`? Seems odd.

Reply:
Yes -- this is expected.  @logand22 can you provide context?